src/protodefc.rs

```diff
@@ -119,6 +116,37 @@ fn run(matches: &clap::ArgMatches) -> Result<()> {
     Ok(())
 }
 
+fn open_input(name: &str) -> File {
+    #[cfg(any(unix))]
+    {
+        use std::os::unix::io::FromRawFd;
+        if (name == "-") {
+            // NOTE: This is unsafe because File expects that it
+            // is the sole user of this descriptor. Any access to
+            // standard input in other locations could cause issues
+            return unsafe { File::from_raw_fd(0) }
```

Comment:
Why are you using `from_raw_fd` instead of `std::io::stdin` and `std::io::stdout`?

Using `from_raw_fd` is both potentially memory unsafe, and this solution only works on unix.